hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala

```diff
@@ -3,93 +3,55 @@ package is.hail.expr.ir.functions
 import is.hail.asm4s
 import is.hail.asm4s._
 import is.hail.expr.ir._
-import is.hail.expr.types.physical.{PBoolean, PCanonicalString, PInt32, PType}
+import is.hail.expr.types.physical.{PBoolean, PCanonicalString, PInt32, PLocus, PType}
 import is.hail.expr.types.virtual._
-import is.hail.utils._
 import is.hail.variant.ReferenceGenome
 
-class ReferenceGenomeFunctions(rg: ReferenceGenome) extends RegistryFunctions {
+object ReferenceGenomeFunctions extends RegistryFunctions {
+  def rgCode(mb: EmitMethodBuilder[_], rg: ReferenceGenome): Code[ReferenceGenome] = mb.getReferenceGenome(rg)
 
-  def rgCode(mb: EmitMethodBuilder[_]): Code[ReferenceGenome] = mb.getReferenceGenome(rg)
-
-  var registered: Set[String] = Set[String]()
-
-  def removeRegisteredFunctions(): Unit =
-    registered.foreach(IRFunctionRegistry.removeIRFunction)
-
-  def registerRGCode(
-    mname: String, args: Array[Type], rt: Type, pt: (Type, Seq[PType]) => PType)(
-    impl: (EmitRegion, PType, Array[(PType, Code[_])]) => Code[_]
-  ): Unit = {
-    val newName = rg.wrapFunctionName(mname)
-    registered += newName
-    registerCode(newName, args, rt, pt)(impl)
-  }
-
-  def registerRGCode[A1](
-    mname: String, arg1: Type, rt: Type, pt: (Type, PType) => PType)(
-    impl: (EmitRegion, PType, (PType, Code[A1])) => Code[_]
-  ): Unit =
-    registerRGCode(mname, Array[Type](arg1), rt, unwrappedApply(pt)) {
-      case (r, rt, Array(a1: (PType, Code[A1]) @unchecked)) => impl(r, rt, a1)
-    }
-
-  def registerRGCode[A1, A2](
-    mname: String, arg1: Type, arg2: Type, rt: Type, pt: (Type, PType, PType) => PType)(
-    impl: (EmitRegion, PType, (PType, Code[A1]), (PType, Code[A2])) => Code[_]
-  ): Unit =
-    registerRGCode(mname, Array[Type](arg1, arg2), rt, unwrappedApply(pt)) {
-      case (r, rt, Array(a1: (PType, Code[A1]) @unchecked, a2: (PType, Code[A2]) @unchecked)) => impl(r, rt, a1, a2)
-    }
-
-  def registerRGCode[A1, A2, A3, A4](
-    mname: String, arg1: Type, arg2: Type, arg3: Type, arg4: Type, rt: Type, pt: (Type, PType, PType, PType, PType) => PType)(
-    impl: (EmitRegion, PType, (PType, Code[A1]), (PType, Code[A2]), (PType, Code[A3]), (PType, Code[A4])) => Code[_]
-  ): Unit =
-    registerRGCode(mname, Array[Type](arg1, arg2, arg3, arg4), rt, unwrappedApply(pt)) {
-      case (r, rt, Array(
-      a1: (PType, Code[A1]) @unchecked,
-      a2: (PType, Code[A2]) @unchecked,
-      a3: (PType, Code[A3]) @unchecked,
-      a4: (PType, Code[A4]) @unchecked)) => impl(r, rt, a1, a2, a3, a4)
+  def registerAll() {
+    registerCode("isValidContig", LocusFunctions.tlocus("R"), TString, TBoolean, (_: Type, _: PType) => PBoolean()) {
```

Comment:
Importing LocusFunctions here doesn't seem right.  I'd move tlocus up to Functions with tv, etc.

Reply:
will issue a small PR to follow up, seems a bit out of scope for this particular PR